application/src/main/java/org/togetherjava/tjbot/commands/moderation/scam/ScamBlocker.java

```diff
@@ -0,0 +1,364 @@
+                .queue();
+            return;
+        }
+
+        MessageUtils.disableButtons(event.getMessage());
+        event.deferEdit().queue();
+        if (event.getButton().getStyle() == ButtonStyle.DANGER) {
+            logger.info(
+                    "Identified a false-positive scam (id '{}', hash '{}') in guild '{}' sent by author '{}'",
+                    args.messageId, args.contentHash, args.guildId, args.authorId);
+            return;
+        }
+        Guild guild = event.getJDA().getGuildById(args.guildId);
+        if (guild == null) {
+            logger.debug(
+                    "Attempted to handle scam, but the bot is not connected to the guild '{}' anymore, skipping scam handling.",
+                    args.guildId);
+            return;
+        }
+
+        Consumer<Member> onRetrieveAuthorSuccess = author -> {
+            quarantineAuthor(guild, author, event.getJDA().getSelfUser());
+            dmUser(guild, args.authorId, event.getJDA());
+
+            // Delete all messages like this
+            Collection<ScamHistoryStore.ScamIdentification> scamMessages = scamHistoryStore
+                .markScamDuplicatesDeleted(args.guildId, args.authorId, args.contentHash);
+
+            scamMessages.forEach(scamMessage -> {
+                TextChannel channel = guild.getTextChannelById(scamMessage.channelId());
+                if (channel == null) {
+                    logger.debug(
+                            "Attempted to delete scam messages, bot the channel '{}' does not exist anymore, skipping deleting messages for this channel.",
+                            scamMessage.channelId());
+                    return;
+                }
+
+                channel.deleteMessageById(scamMessage.messageId()).mapToResult().queue();
+            });
+        };
+
+        Consumer<Throwable> onRetrieveAuthorFailure = failure -> {
+            if (failure instanceof ErrorResponseException errorResponseException) {
+                if (errorResponseException.getErrorResponse() == ErrorResponse.UNKNOWN_USER) {
+                    logger.debug("Attempted to handle scam, but user '{}' does not exist anymore.",
+                            args.authorId);
+                    return;
+                }
+
+                if (errorResponseException.getErrorResponse() == ErrorResponse.UNKNOWN_MEMBER) {
+                    logger.debug(
+                            "Attempted to handle scam, but user '{}' is not a member of the guild anymore.",
+                            args.authorId);
+                }
+            }
+        };
+
+        guild.retrieveMemberById(args.authorId)
+            .queue(onRetrieveAuthorSuccess, onRetrieveAuthorFailure);
+    }
+
+
+    private record ComponentIdArguments(@NotNull ScamBlockerConfig.Mode mode, long guildId,
+            long channelId, long messageId, long authorId, String contentHash) {
+        static ComponentIdArguments fromList(@NotNull List<String> args) {
```

Comment:
```suggestion
            long channelId, long messageId, long authorId, String contentHash) {
            
        static ComponentIdArguments fromList(@NotNull List<String> args) {
```
I thought this was the constructor, got really confused lol